src/main/mocha.ts

```diff
@@ -162,89 +168,90 @@ export default class MochaWrapper extends Mocha {
   }
 
   private spawnTestProcess(file: string): Promise<ISubprocessResult> {
-    return new Promise((resolve) => {
-      const nodeFlags: string[] = [];
-      const extension = this.isTypescriptRunMode ? 'ts' : 'js';
-      const runnerPath = pathResolve(__dirname, `../subprocess/runner.${extension}`);
+    return new Promise<ISubprocessResult>(async (resolve) => {
       const resolvedFilePath = pathResolve(file);
 
-      const forkArgs: string[] = ['--test', resolvedFilePath];
+      const testOptions: {[key: string]: any} = { test: resolvedFilePath };
+
       for (const option of SUITE_OWN_OPTIONS) {
         const propValue = this.suite[option]();
         // bail is undefined by default, we need to somehow pass its value to the subprocess
-        forkArgs.push(`--${option}`, propValue === undefined ? false : propValue);
+        testOptions[option] = propValue === undefined ? false : propValue;
       }
 
       for (const requirePath of this.requires) {
-        forkArgs.push('--require', requirePath);
+        testOptions.require = requirePath;
       }
 
-      for (const compilerPath of this.compilers) {
-        forkArgs.push('--compilers', compilerPath);
-      }
+      testOptions.compilers = this.compilers || [];
 
       if (this.options.delay) {
-        forkArgs.push('--delay');
+        testOptions.delay = true;
       }
 
       if (this.options.grep) {
-        forkArgs.push('--grep', this.options.grep.toString());
+        testOptions.grep = this.options.grep.toString();
       }
 
       if (this.exitImmediately) {
-        forkArgs.push('--exit');
+        testOptions.exit = true;
       }
 
       if (this.options.fullStackTrace) {
-        forkArgs.push('--full-trace');
+        testOptions.fullStackTrace = true;
       }
 
-      const test = fork(runnerPath, forkArgs, {
-        // otherwise `--inspect-brk` and other params will be passed to subprocess
-        execArgv: process.execArgv.filter(removeDebugArgs),
-        stdio: ['ipc'],
-      });
-
-      if (this.isTypescriptRunMode) {
-        nodeFlags.push('--require', 'ts-node/register');
+      let test;
+      try {
+        test = await this.pool.getOrCreate(this.isTypescriptRunMode);
+      } catch (e) {
+        throw e;
       }
 
-      debugLog('Process spawned. You can run it manually with this command:');
-      debugLog(`node ${nodeFlags.join(' ')} ${runnerPath} ${forkArgs.concat([DEBUG_SUBPROCESS.argument]).join(' ')}`);
+      test.send(JSON.stringify({ type: 'start', testOptions }));
```

Comment:
Minor: it looks like `getOrCreate` returns a node.js process. What do you think of returning some abstraction object instead of it? Operating with actual IPC `send` here (not in the `process-pool.ts` file) looks a bit low-level to me.

Reply:
Switched to a simple handle.